vnext/proposals/active/keyboard-reconcile-desktop.md

```diff
@@ -0,0 +1,148 @@
+| onKeyDownCapture | IKeyboardEvent | void | Occurs when the onKeyDown event is being routed. onKeyDown is the corresponding bubbling event. On Windows, this corresponds to [PreviewKeyDown](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.previewkeydown) |
+| onKeyUp | IKeyboardEvent | void | Occurs when a keyboard key is released when a component has focus. On Windows, this corresponds to [KeyUp](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.keyup) |
+| onKeyUpCapture | IKeyboardEvent | void | Occurs when the onKeyUp event is being routed. onKeyUp is the corresponding bubbling event. On Windows, this corresponds to [PreviewKeyUp](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.previewkeyup) |
+
+Where `IKeyboardEvent` is a new event type added to `ReactNative.NativeSyntheticEvents` of type `INativeKeyboardEvent`. The properties in [NativeSyntheticEvent](https://reactjs.org/docs/events.html#overview) like target, bubbles, cancelable etc., are also available for `IKeyboardEvent` and follow the same behaviors as other events in react-native today.
+
+`INativeKeyboardEvent` is a new interface and will expose the following properties:
+
+| **Property** | **Type** | **Description** | **Default** |
+| --- | --- | --- | --- |
+| key | string | The character typed by the user.TODO: Document the w3c spec for how the keys show up | string.Empty |
+| altKey | boolean | The Alt (Alternative) key. Also maps to Apple Option key. | false |
+| ctrlKey | boolean | The Ctrl (Control) key. | false |
+| shiftKey | boolean | The Shift key. | false |
+| metaKey | boolean | Maps to Windows Logo key and the Apple Command key. | False |
+| repeat | boolean | Flag to represent if a key is being held down/repeated. Tracked by [Windows Issue#5513](https://github.com/microsoft/react-native-windows/issues/5513), [macOS Issue#502](https://github.com/microsoft/react-native-macos/issues/502) | False |
+| ~~eventPhase~~ | ~~EventPhase~~ | ~~Current phase of routing for the key event.~~ | ~~Bubbling~~ |
+
+~~Where EventPhase is an enum to detect whether the keystroke is being tunneled/bubbled to the target component that has focus. It has the following fields:~~
+
+- ~~None : none~~
+- ~~Capturing : when the keydown/keyup event is being captured while tunneling its way from the root to the target component~~
+- ~~AtTarget : when the keydown/keyup event has reached the target component that is handling the corresponding event~~
+- ~~Bubbling : when the keydown/keyup event is being captured while bubbling its way to the parent(s) of the target component~~
+
+In the following example, the lastKeyDown prop will contain the key stroke from the end user when keyboard focus is on View.
+```
+  <View onKeyDown={this._onKeyDown} />
+      
+  private _onKeyDown = (event: IKeyboardEvent) => {
+    this.setState({ lastKeyDown: event.nativeEvent.key });
+  };
+
+```
+
+## Declarative properties
```

Comment:
Is this equivalent to calling "preventDefault" on the web? If so, it seems that Fabric should avoid the need for this part of the proposal. In which case, is this an interim API for use while Fabric is still not widely available? And what could we do to encourage forward-looking use (e.g., alongside these declarative properties we enforce calling `preventDefault` on the event even though it won't do anything without a Fabric-backed implementation)?

Reply:
@acoates-ms , @FalseLobster  - can you comment on the thinking here?

Reply:
Yes this is the equivalent to calling preventDefault.  But I don't think that Fabric prevents the need for this.  I don't think the plan is for Fabric to handle all event synchronously on the UI thread. -- That would be quite a departure from how RN works today, and likely to cause performance issues.

Reply:
Internally I think there is interest in doing this to avoid the declarative properties. Scott may have more to say. Not all events would need to support preventDefault though. The web has shifted to making a lot of events passive by default, and passive events don't support preventDefault.

Reply:
Yep, Fabric will have support for synchronous event dispatching. This would likely be a perf win for keyboard events because it would reduce the lag between keyboard input and updating the screen, and it will allow us to deprecate these declarative properties.

I agree with @necolas that it would be good to trigger a warning if `preventDefault()` isn't called when it's expected to be, as that would make the deprecation go smoother.

Reply:
I like the idea of forcing them to call preventDefault to match the declarative properties.

Reply:
Sounds righteous. Filed #5822 to track this work.